File: python3_solution_set2/2383. Minimum Hours of Training to Win a Competition.py
class Solution:
    def minNumberOfHours(self, initialEnergy: int, initialExperience: int, energy: List[int], experience: List[int]) -> int:
        
        eng = 0
        exp = 0
        ## experience gap
        for x in experience:
            if x < initialExperience:
                initialExperience+=x
            else:
                exp+=x-initialExperience+1
                initialExperience+=x-initialExperience+1
                initialExperience+=x
        
        ## energy gap
        if sum(energy) < initialEnergy:
            eng = 0
        else:
            eng+=sum(energy) - initialEnergy + 1
        #print(eng,exp)
        return eng + exp

File: cpp_solutions/2383. Minimum Hours of Training to Win a Competition.cc
#include <numeric>
class Solution {
public:
    int minNumberOfHours(int initialEnergy, int initialExperience, vector<int>& energy, vector<int>& experience) {
        int eng = 0;
        int exp = 0;
        
        // energy training
        if (accumulate(energy.begin(),energy.end(),0) < initialEnergy) {
            eng = 0;
        } else {
            eng+= accumulate(energy.begin(),energy.end(),0) - initialEnergy + 1;
        }
        
        // experience training
        for (auto it : experience) {
            if (it < initialExperience) {
                initialExperience+=it;
            } else {
                exp+=it - initialExperience+1;
                initialExperience += it - initialExperience + 1;
                initialExperience+= it;
            }
        }
        
        
        
        return eng + exp;
        
    }
};
